src/buildApplication/fixPrerenderedRoutes.ts

```diff
@@ -0,0 +1,208 @@
+import { cliWarn } from '../cli';
+import {
+	formatRoutePath,
+	normalizePath,
+	readJsonFile,
+	stripIndexRoute,
+	validateFile,
+} from '../utils';
+import { join, relative, resolve, dirname } from 'path';
+import { mkdir, copyFile } from 'fs/promises';
+
+export type VercelPrerenderConfig = {
+	type: string;
+	sourcePath?: string;
+	fallback: { type: string; mode: number; fsPath: string };
+	initialHeaders?: Record<string, string>;
+};
+
+export type PrerenderedFileData = {
+	headers?: Record<string, string>;
+	overrides?: string[];
+};
+
+/**
+ * Retrieve a valid prerendered route config.
+ *
+ * @param baseDir Base directory for the prerendered routes.
+ * @param file Prerendered config file name.
+ * @param dirName Directory name to use for the route.
+ * @returns A valid prerendered route config, or null if the config is invalid.
+ */
+async function getRouteConfig(
+	baseDir: string,
+	file: string,
+	dirName: string
+): Promise<VercelPrerenderConfig | null> {
+	const configPath = join(baseDir, file);
+	const config = await readJsonFile<VercelPrerenderConfig>(configPath);
+
+	if (
+		config?.type?.toLowerCase() !== 'prerender' ||
+		config?.fallback?.type?.toLowerCase() !== 'filefsref' ||
+		!config?.fallback?.fsPath
+	) {
+		const relativeName = normalizePath(join(dirName, file));
+		cliWarn(`Invalid prerender config for ${relativeName}`);
+		return null;
+	}
+
+	return config;
+}
+
+/**
+ * Retrieve the path to the prerendered route, if it exists.
+ *
+ * @param config.fallback Fallback file configuration.
+ * @param dirName Directory name to use for the route.
+ * @param outputDir Vercel build output directory.
+ * @returns The path to the prerendered route, or null if it does not exist.
+ */
+async function getRoutePath(
+	{ fallback }: VercelPrerenderConfig,
+	dirName: string,
+	outputDir: string
+): Promise<string | null> {
+	const oldRoute = normalizePath(join(dirName, fallback.fsPath));
+	const oldFile = join(outputDir, 'functions', oldRoute);
+
+	// Check the prerendered file exists.
+	if (!(await validateFile(oldFile))) {
+		cliWarn(`Could not find prerendered file for ${oldRoute}`);
+		return null;
+	}
+
+	return oldFile;
```

Comment:
If I read this function without external context I find the prefixes `old` rather unclear/confusing, would you mind removing them 😓 🙏 
```suggestion
	const prerenderRoute = normalizePath(join(dirName, fallback.fsPath));
	const prerenderFile = join(outputDir, 'functions', prerenderRoute);

	// Check the prerendered file exists.
	if (!(await validateFile(prerenderFile))) {
		cliWarn(`Could not find prerendered file for ${prerenderRoute}`);
		return null;
	}

	return prerenderFile;
```